test/integration/instance.validations.test.js

```diff
@@ -317,6 +317,37 @@ describe(Support.getTestDialectTeaser('InstanceValidator'), function() {
           });
         });
       });
+
+      it('should enforce a unque constraint', function() {
+        var Model = this.sequelize.define('model', {
+          uniqueName: { type: Sequelize.STRING, unique: true }
+        });
+        var records = [
+          { uniqueName: 'unique name one' },
+          { uniqueName: 'unique name two' }
+        ];
+        var expectedId = -1;
+        return Model.sync({ force: true })
+          .then(function() {
+            return Model.create(records[0]);
+          }).then(function(instance) {
+            expect(instance).to.be.ok;
+            return Model.create(records[1]);
+          }).then(function(instance) {
+            expect(instance).to.be.ok;
+            expectedId = instance.id;
+            return Model.update(records[0], { where: { id: instance.id } });
+          }).then(function(instance) {
```

Comment:
The `throw` part looks a bit smelly to me. I'd change it to:

```js
.then(function () {
  return expect(Model.update(records[0], { where: { id: instance.id } })).to.be.rejected;
}).then(function (err) {
  // further assertions about err here
});
``` 

Expect to be rejected will add a catch handler and return the error in the `then` callback